docs/Changelog.md

```diff
@@ -24582,6 +24582,72 @@ This version of the operator has been available since version 21 of the default
 <dd>'x_scale' determines the output type.</dd>
 </dl>
 
+### <a name="GroupNormalization-21"></a>**GroupNormalization-21**</a>
+
+  A GroupNormalization function. Carries out group normalization as described in
+  the paper https://arxiv.org/abs/1803.08494
+
+  This operator transforms input according to
+  ```
+  y = scale * (x - mean) / sqrt(variance + epsilon) + bias,
+  ```
+  where the mean and variance are computed per instance per group of channels, and
+  `scale` and `bias` should be specified for each group of channels. The number of
+  groups `num_groups` should be divisible by the number of channels so that there are
+  an equal number of channels per group.
+
+  The overall computation has two stages: the first stage normalizes the elements to
+  have zero mean and unit variance for each instance in each group, and the second
+  stage scales and shifts the results of the first stage. The floating-point precision
+  used in the first stage is determined by the `stash_type` attribute. For example,
+  if `stash_type` is 1, the operator casts all input variables to 32-bit float,
+  performs the computation, and finally casts the normalized results back to the
+  original type of `X`. The second stage does not edepend on `stash_type`.
+
+  When the number of groups is the same as the number of channels, this operator is
+  equivalent to InstanceNormalization. When there is only one group, this operator
+  is equivalent to LayerNormalization.
+
+#### Version
+
+This version of the operator has been available since version 21 of the default ONNX operator set.
+
+#### Attributes
+
+<dl>
+<dt><tt>epsilon</tt> : float (default is 1e-05)</dt>
+<dd>The epsilon value to use to avoid division by zero.</dd>
+<dt><tt>num_groups</tt> : int (required)</dt>
```

Comment:
PyTorch has the `num_channels` parameter (https://pytorch.org/docs/stable/generated/torch.nn.GroupNorm.html) do you see it being supported?

Reply:
`num_channels` can be inferred from the input shape already - that's why it wasn't included here.
We can consider including it for better conformity with torch. Do you have a preference? cc: @gramalingam

Reply:
If it is inferred I wouldn't bother adding that, thanks!

Reply:
Agree. I don't think we should add it unless there is a good reason. May be Pytorch needs it to allocate the scale/bias tensors that are learnt via training? For us, they (scale/bias) are extra inputs.